src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -487,6 +487,17 @@ protected virtual string GetInsertionText(CompletionItem item, char ch)
         // This is used to decide which provider we'd collect target type completion telemetry from.
         protected virtual bool ShouldCollectTelemetryForTargetTypeCompletion => false;
 
+        protected static CompletionChange GetCompletionChangeWithParenthesis(
+            string insertionText, char commitKey, TextSpan itemSpan)
+        {
+            var text = string.Concat(insertionText + "()", commitKey);
```

Comment:
In the context for invocation doesn't mean we can always append a ";" after the invocation expression., this might be OK since it's what user typed, but I think it'd be a better experience if we make a smart decision based on context. Thoughts?

Reply:
Emm, do you mean we should override the GetInsertionText method and let the commit manager if it wants to call other completion service for commit key? Or we should make a decision here based on the syntax & semantic context?